gh-126083: Fix a reference leak in `asyncio.Task` when reinitializing with new non-`None` context (#126103)
python/cpython: Misc/NEWS.d/next/Library/2024-10-28-22-35-22.gh-issue-126083.TuI--n.rst

> Fixed a reference leak that would happen in asyncio tasks when you would
> reinitialize your task object with a non-None context.

Reviewer: ```suggestion
Fixed a reference leak in :class:`asyncio.Task`
when reinitializing the same object with a non ``None`` context.
```

Reviewer 2: Completing Nikita's suggestion:

```suggestion
Fixed a reference leak in :class:`asyncio.Task` objects
when reinitializing the same object with a non-``None`` context.
```

You are also welcome to add your name / pseudonym via "Patch by [...]"!
